src/modules/launcher/Plugins/Microsoft.PowerToys.Run.Plugin.OneNote/Main.cs

```diff
@@ -0,0 +1,150 @@
+using Windows.Win32;
+using Windows.Win32.Foundation;
+using Windows.Win32.UI.WindowsAndMessaging;
+using Wox.Plugin;
+
+namespace Microsoft.PowerToys.Run.Plugin.OneNote
+{
+    /// <summary>
+    /// A power launcher plugin to search across time zones.
+    /// </summary>
+    public class Main : IPlugin, IPluginI18n
+    {
+        /// <summary>
+        /// A value indicating if the OneNote interop library was able to successfully initialize.
+        /// </summary>
+        private readonly bool _oneNoteInstalled;
+
+        /// <summary>
+        /// The initial context for this plugin (contains API and meta-data)
+        /// </summary>
+        private PluginInitContext? _context;
+
+        /// <summary>
+        /// The path to the icon for each result
+        /// </summary>
+        private string _iconPath;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Main"/> class.
+        /// </summary>
+        public Main()
+        {
+            UpdateIconPath(Theme.Light);
+            try
+            {
+                _ = OneNoteProvider.PageItems.Any();
+                _oneNoteInstalled = true;
+            }
+            catch (COMException)
+            {
+                // OneNote isn't installed, plugin won't do anything.
+                _oneNoteInstalled = false;
+            }
+        }
+
+        /// <summary>
+        /// Gets the localized name.
+        /// </summary>
+        public string Name => Resources.PluginTitle;
+
+        /// <summary>
+        /// Gets the localized description.
+        /// </summary>
+        public string Description => Resources.PluginDescription;
+
+        /// <summary>
+        /// Initialize the plugin with the given <see cref="PluginInitContext"/>
+        /// </summary>
+        /// <param name="context">The <see cref="PluginInitContext"/> for this plugin</param>
+        public void Init(PluginInitContext context)
+        {
+            _context = context ?? throw new ArgumentNullException(nameof(context));
+
+            _context.API.ThemeChanged += OnThemeChanged;
+            UpdateIconPath(_context.API.GetCurrentTheme());
+        }
+
+        /// <summary>
+        /// Return a filtered list, based on the given query
+        /// </summary>
+        /// <param name="query">The query to filter the list</param>
+        /// <returns>A filtered list, can be empty when nothing was found</returns>
+        public List<Result> Query(Query query)
+        {
+            if (!_oneNoteInstalled || query is null || string.IsNullOrWhiteSpace(query.Search))
+            {
+                return new List<Result>(0);
+            }
+
+            var pages = OneNoteProvider.FindPages(query.Search);
+
+            return pages.Select(p => new Result
+            {
+                IcoPath = _iconPath,
+                Title = p.Name,
+                QueryTextDisplay = p.Name,
+                SubTitle = @$"{p.Notebook.Name}\{p.Section.Name}",
```

Comment:
Os the following possible?

```suggestion
                SubTitle = @$"{p.path}\{p.Notebook.Name}:{p.Section.Name}",
```

Reply:
It's possible to display the p.Notebook.Path, but it's quite a long URL usually, assuming it's in sharepoint/onedrive, etc, so I wouldn't personally suggest showing it

Reply:
Help me a bit here. Is `p` the page or file? Or is file = page? I am not sure if it's possible to have the same page name in different books. (If yes we get a problem with the result navigation as you can't navigate by courser between results with the same title in Run's ui.)

Reply:
p is a page. The hierarchy is Notebook -> [Section Group ->] Section -> Page, but the file is at the Notebook level. It is possible to have the same page name in different notebooks, but I just tested it and it works fine as-is, I was able to navigate to the different pages w/o issue